nomos-da/network/core/src/addressbook/mock.rs

```diff
@@ -0,0 +1,29 @@
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+
+use libp2p::{Multiaddr, PeerId};
+
+use crate::addressbook::{AddressBookHandler, AddressBookMut};
+
+#[derive(Default, Debug, Clone)]
+pub struct MockAddressBook {
+    peers: Arc<Mutex<HashMap<PeerId, Multiaddr>>>,
+}
+
+impl AddressBookHandler for MockAddressBook {
```

Comment:
`nomos-da-network-core` is proper place for the Trait definition, maybe it could be moved just to `addressbook.rs` or left as is, but Mock and other implementations shouldn't be here in my opinion. 

Just like MembershipHandler, we should have the implementation in DaNetwork service, real implementation will have to communicate with the storage.
For tests in nomos-da-network-core we could just redefine it in test_utils

Reply:
That sounds better to me as well, I will move it as suggested. Thanks!